index.bs

```diff
@@ -162,13 +165,26 @@ The <dfn attribute for="XRInputSource">gamepad</dfn> attribute is a {{Gamepad}}
  - More than one button
  - One or more axes
 
-Note: {{XRInputSource}}s in an {{XRSession}}'s {{XRSession/inputSources}} array are "live". As such values within them, such as the {{gamepad}} button and axis state, are updated in-place. This means that it doesn't work to save a reference to an {{XRInputSource}}'s {{gamepad}} on one frame and compare it to the same {{XRInputSource}}'s {{gamepad}} from a subsequent frame to test for state changes, because they will be the same object. Therefore developers that wish to compare input state from frame to frame should copy the state in question.
-
 XRSession {#xrsession-interface}
 ------------
 
 {{XRInputSource}}s are reported in the {{XRSession/inputSources}} array as they are connected and disconnected. When the presence of a {{XRInputSource/gamepad}} changes for any entry in the {{XRSession/inputSources}} array, the user agent MUST invoke the WebXR Device API's algorithm for [=change input source|responding to input source attribute changes=].
 
+XRFrame {#xrframe-interface}
+------------
+
+The [=XRFrame/apply frame updates=] steps are modified to include [=apply gamepad frame updates=].
+
+To <dfn for=XRFrame>apply gamepad frame updates</dfn> for an {{XRFrame}} |frame|, the user agent MUST run the following steps:
+
+ 1. For each {{XRInputSource}} with a {{XRInputSource/gamepad}} |gamepad| associated with |frame|'s {{XRFrame/session}}, perform the following steps:
+    1. Update |gamepad| to reflect the gamepad state at |frame|'s [=XRFrame/time=].
```

Comment:
Is there an update algorithm we can refer to from the Gamepad API?  Because this is a bit vauge.  Which properties will be changed and where does the data come from?

Reply:
Nah, part of the whole problem here is that Gamepad doesn't talk about liveness at all, so this stuff is already super vague and implicit there. The gamepad spec uses the word "data" a lot instead of "state", I've changed this to use that, but it's not a dfn or used super consistently.

Once https://github.com/w3c/gamepad/issues/8 is fixed we shall probably have better tools for talking about this.